hist/hist/src/TKDE.cxx

```diff
@@ -563,22 +595,38 @@ void TKDE::SetSigma(Double_t R) {
 
 void TKDE::SetKernel() {
    // Sets the kernel density estimator
-   UInt_t n = fData.size();
+   // n here should be fNEvents or the total size including the mirrored events ???
+   // it should not be fData.size() that in binned case is number of bins
+   UInt_t n =  (fUseBins) ? fNBins : fNEvents;
    if (n == 0) return;
    // Optimal bandwidth (Silverman's rule of thumb with assumed Gaussian density)
    Double_t weight = fCanonicalBandwidths[kGaussian] * fSigmaRob * std::pow(3. / (8. * std::sqrt(M_PI)) * n, -0.2);
    weight *= fRho * fCanonicalBandwidths[fKernelType] / fCanonicalBandwidths[kGaussian];
-   if (fKernel) delete fKernel;
-   fKernel = new TKernel(weight, this);
+
+   fKernel.reset( new TKernel(weight, this) );
```

Comment:
I think in this situation, `fKernel = std::make_unique<TKernel>(weight, this)` should be preferred. At least that's what I read in the core guidelines:

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#r23-use-make_unique-to-make-unique_ptrs

Reply:
I agree, make_unique is preferable, but I think is from C++-14 and not 11.